lib/cisco_node_utils/span_session.rb

```diff
@@ -0,0 +1,125 @@
+# Insert appropriate license here
+require_relative 'node_util'
+require_relative 'interface'
+
+
+module Cisco
+  # node utils class for SPAN (switchport analyzer) sessions
+  class SpanSession < NodeUtil
+    attr_reader :session_id
+
+    def initialize(session_id)
+      validate_args(session_id)
+    end
+
+    def self.sessions
+      hash = {}
+      all = config_get('span_session', 'all_sessions')
+      return hash if all.nil?
+
+      all.each do |id|
+        id = id.downcase
+        hash[id] = SpanSession.new(id)
+      end
+      hash
+    end
+
+    def validate_args(session_id)
+      fail TypeError unless session_id.is_a?(Integer)
+      fail ArgumentError unless session_id >= 1 || session_id <= 32
+      @session_id = session_id.downcase
```

Comment:
Using `downcase` here will generate an error since `session_id` is a `Fixnum`.  `@session_id = session_id` should be sufficient.